src/legacy/core_plugins/kibana/public/management/saved_object_registry.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import _ from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { npStart } from 'ui/new_platform';
```

Comment:
you shouldn't be using npStart in this files but rather pass everything needed in as a dependency, to save you some problems later on when you try to move to the new platform. 

take a look at `plugins/data/public/services.ts`, you probably want to create a file like this inside your plugins, in the `plugin.ts` you would call the setters and then in the items you register to the registry you can call the getters.

this allows to provide `setup` and `start` dependencies to the items you register, assuming that nobody will be using the items after all the plugins had a chance to start (usually by returning methods to read the registry only in the start method)

Reply:
AFAIK this file is not expected to make it into the new platform. Platform team will look into this, but I talked with @rudolf about this a while ago and back then the plan was to get rid of this registry completely.

Reply:
i would still prefer to see this updated,
- just in case we at the end decide to take this to the new platform (other code is relying on this)
- to avoid any weird errors, which quite often pop up when you are shimming something, due to circular imports
- to follow the conventions we follow in other places

and it shouldn't take much work. @kertal its up to you :)

Reply:
@lizozom asked why this registry was used in the timelion part of Kibana. It's not needed and I've removed it. This was the last part outside the saved object management it is needed. So there is no danger of circular imports. There is currently no `plugin.ts` in this section, so the part of migrating it to NP structure should be done in a separate step.